### PR TITLE
Adding algorithm for finding diameter of a binary tree

### PR DESCRIPTION
**Pull request by shreyneil**
**Fixes issue:** https://github.com/OpenGenus/cosmos/issues/1962

**Changes:**
I have implemented the algorithm for finding diameter of a binary tree in c++. Please go through the code and suggest changes if any.

### CODE REVIEW

File: code/data_structures/tree/binary_tree_diameter/b_tree_diameter.cpp [hunk 1]
```diff
@@ -0,0 +1,69 @@
+#include<iostream>
+using namespace std;
+
```
Comment (Vitorvgc) [inline code/data_structures/tree/binary_tree_diameter/b_tree_diameter.cpp]: Don't forget to add the required comment as stated in the [contribution guidelines](https://github.com/OpenGenus/cosmos/wiki/contribute).

Comment (shreyneil) [reply code/data_structures/tree/binary_tree_diameter/b_tree_diameter.cpp]: Done.

Comment (Vitorvgc) [reply code/data_structures/tree/binary_tree_diameter/b_tree_diameter.cpp]: Aren't you missing the `/* */` ?

Comment (shreyneil) [reply code/data_structures/tree/binary_tree_diameter/b_tree_diameter.cpp]: Done.

**Comment by AdiChat** [comment]
Looks great 😍 

Hope you are enjoying your journey 🚆  with **OpenGenus**

**Pull request merged**